File: ictyb_front/app/routes/qna/list/QnaDto.ts
// Q&A (공지사항 테이블 기반) 타입
export interface QnaItem {
  noticeNo: number;
  noticeTitle: string;
  noticeDepCd: string;
  noticeContents: string;
  priority: number;
  regUserSabun: string;
  regUserDepCd: string;
  regUserName: string;
  regDt: string;
  endDt: string;
  delYn: string;
  viewCnt: number;
  noticeType: string; // 'Q' = Q&A
}

// ─── Q&A List 조회 ────────────────────────────────────────────────────
export interface QnaListItem {
  noticeNo: number;
  noticeTitle: string;
  regUserName: string;
  regDt: string;
  viewCnt: number;
  attachCount: number;
}


// ─── Q&A 등록 폼 ─────────────────────────────────────────────────
export interface QnaRegisterForm {
  noticeTitle: string;
  noticeContents: string;
  noticeDepCd: string;
  regUserSabun: string;
  regUserDepCd: string;
  regUserName: string;
  endDt: string;
  priority: number;
}


// ─── Q&A 상세 ────────────────────────────────────────────────────
export interface QnaAttach {
  noticeNo: string;
  seq: string;
  realFileName: string;
  fileName: string;
  fileLocation: string;
  regDt: string;
  attachType: string;
  fileSize: string;
}

export interface QnaDetail {
  noticeNo: number;
  noticeTitle: string;
  noticeDepCd: string;
  noticeContents: string;
  priority: number;
  regUserSabun: string;
  regUserDepCd: string;
  regUserName: string;
  regDt: string;
  endDt: string;
  delYn: string;
  viewCnt: number;
  noticeType: string;
  attachList: QnaAttach[];
}


export interface QnaDetail extends QnaItem {
  attachList: QnaAttach[];
}

File: ictyb_back/src/main/java/kepco/prorject/ictyb/back/ictyb_back/qna/model/QnaDto.java
package kepco.prorject.ictyb.back.ictyb_back.qna.model;


import lombok.*;

import java.util.List;

public class QnaDto {

    // ── 목록 응답 ─────────────────────────────────────────────────
    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class ListItem {
        private Long   noticeNo;
        private String noticeTitle;
        private String regUserName;
        private String regDt;
        private int    viewCnt;
        private int    attachCount;
    }

    // ── 상세 응답 ─────────────────────────────────────────────────
    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class Detail {
        private Long         noticeNo;
        private String       noticeTitle;
        private String       noticeDepCd;
        private String       noticeContents;
        private int          priority;
        private String       regUserSabun;
        private String       regUserDepCd;
        private String       regUserName;
        private String       regDt;
        private String       endDt;
        private String       delYn;
        private int          viewCnt;
        private String       noticeType;
        private List<Attach> attachList;
    }

    // ── 첨부파일 ──────────────────────────────────────────────────
    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class Attach {
        private String noticeNo;
        private String seq;
        private String realFileName;
        private String fileName;
        private String fileLocation;
        private String regDt;
        private String attachType;
        private String fileSize;
    }

    // ── 등록 요청 ─────────────────────────────────────────────────
    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    public static class RegisterRequest {
        private String noticeTitle;
        private String noticeContents;
        private String noticeDepCd;
        private String regUserSabun;
        private String regUserDepCd;
        private String regUserName;
        private String endDt;
        private int    priority;
    }
}
